test/tests/desktop/db/OfflineDbFacadeTest.ts

```diff
@@ -62,4 +63,37 @@ o.spec("OfflineDbFacade", function () {
 
 		verify(factory.create(userId, databaseKey), {times: 2})
 	})
+
+	o("ranges database is locked when writing/reading to/from it", async function () {
+		const listId = "listId"
+
+		// Hold the lock for the ranges database until @param defer is resolved.
+		async function holdRangesDbLock(defer: DeferredObject<void> | null, startId: number): Promise<number> {
+			await offlineDbManager.lockRangesDbAccess(listId)
+			await defer?.promise
+			await offlineDbManager.unlockRangesDbAccess(listId)
+			return startId
+		}
+
+		const finishOrder: Array<number> = []
+		const longRunningTask1 = defer<void>()
+
+		// Task 1
+		holdRangesDbLock(longRunningTask1, 1).then((startId) => {
+			finishOrder.push(startId)
+		})
+		// Task 2
+		holdRangesDbLock(null, 2).then((startId) => {
+			finishOrder.push(startId)
+		})
+
+		// Delay Task 1
+		setTimeout(async () => {
```

Comment:
your test will stop before this will run because your async function will return. you need to either await something. I would pass a `Promise` instead of `DeferredObject` into `holdRangesDbLock()` and then pass `delay(5000)` into one of them? and then do something like `await Promise.all([task1, tas2])` before the verification.

Reply:
Thanks, I will change that :+1: